mautic/mautic
# TPROD-455 Enable PHPStan Doctrine

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [X]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

This PR is part of a sequence of PRs to prepare for Doctrine upgrade, the main goal of this PR is to introduce `phpstan/phpstan-doctrine` which allow us to have a tighter control over entities and repositories.

In the past many of `phpstan` objections were neglected, this PR should solve many of them.   





#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Only code review is required if CI is green
3. **THIS PR REQUIRE CODE REVIEW ONLY**
